test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -260,19 +278,15 @@ public InternalTestCluster(String nodeMode, long clusterSeed, Path baseDir,
         assert nodePrefix != null;
         this.mockPlugins = mockPlugins;
 
-        /*
-         *  TODO
-         *  - we might want start some master only nodes?
-         *  - we could add a flag that returns a client to the master all the time?
-         *  - we could add a flag that never returns a client to the master
-         *  - along those lines use a dedicated node that is master eligible and let all other nodes be only data nodes
-         */
-        sharedNodesSeeds = new long[numSharedAllRolesNodes + numShareCoordOnlyNodes];
+        sharedNodesSeeds = new long[numSharedMasterNodes + numSharedDataNodes + numShareCoordOnlyNodes];
         for (int i = 0; i < sharedNodesSeeds.length; i++) {
             sharedNodesSeeds[i] = random.nextLong();
         }
 
-        logger.info("Setup InternalTestCluster [{}] with seed [{}] using [{}] data nodes and [{}] client nodes", clusterName, SeedUtils.formatSeed(clusterSeed), numSharedAllRolesNodes, numShareCoordOnlyNodes);
+        logger.info("Setup InternalTestCluster [{}] with seed [{}] using [{}] dedicated masters, " +
+                "[{}] (data) nodes and [{}] coord nodes",
```

Comment:
thanks for rephrasing from clients nodes, maybe log instead coord only nodes though?